datafusion/sql/src/unparser/expr.rs

```diff
@@ -151,6 +151,36 @@ impl Unparser<'_> {
                     order_by: vec![],
                 }))
             }
+            Expr::ScalarSubquery(subq) => {
```

Comment:
I have found that all of the information required to preserve a subquery is encoded in the subquery expression. This means we can actually completely ignore subquery LogicalPlan nodes and not lose anything...

At least this is true in every example I have come up with so far :smile:

Reply:
I am not quite sure what you mean by "ignore `LogicalPlan`" as the Expr has a `LogicaPlan` in it, doesn't it (`sub.subquery`)?

Reply:
If you look at `Unparser::select_to_sql_recursively`, we have not implemented a match condition for `LogicalPlan::Subquery`. This node is never actually encountered as I have not found a case where you can get to it via the "input" of another node. We do encounter the Expr::ScalarSubquery expression and we encode the information from the `LogicalPlan` found there instead.

Reply:
Interesting, I wonder if `LogicalPlan:Subquery` is ever actually used 🤔 I took a quick look around and didn't see an obvious creation in the planner.